base/dcp_feed.go

```diff
@@ -463,33 +543,17 @@ func StartDCPFeed(bucket Bucket, spec BucketSpec, args sgbucket.FeedArguments, c
 		persistCheckpoints = true
 	}
 
-	dcpReceiver := NewDCPReceiver(callback, bucket, maxVbno, persistCheckpoints)
-	dcpReceiver.SetBucketNotifyFn(args.Notify)
-
-	// GetStatsVbSeqno retrieves high sequence number for each vbucket, to enable starting
-	// DCP stream from that position.  Also being used as a check on whether the server supports
-	// DCP.
-
-	switch args.Backfill {
-	case sgbucket.FeedNoBackfill:
-		// For non-backfill, use vbucket uuids, high sequence numbers
-		statsUuids, highSeqnos, err := bucket.GetStatsVbSeqno(maxVbno, false)
-		if err != nil {
-			return errors.New("Error retrieving stats-vbseqno - DCP not supported")
-		}
-		LogTo("Feed+", "Seeding seqnos: %v", highSeqnos)
-
-		dcpReceiver.SeedSeqnos(statsUuids, highSeqnos)
-	case sgbucket.FeedResume:
-		// For resume case, load previously persisted checkpoints from bucket
+	dcpReceiver, err := NewDCPReceiver(callback, bucket, maxVbno, persistCheckpoints, args.Backfill)
+	if err != nil {
+		return err
+	}
 
-		dcpReceiver.initMetadata(maxVbno)
-	default:
+	dcpReceiver.SetBucketNotifyFn(args.Notify)
 
-		// Otherwise, start feed from zero
-		startSeqnos := make(map[uint16]uint64, maxVbno)
-		vbuuids := make(map[uint16]uint64, maxVbno)
-		dcpReceiver.SeedSeqnos(vbuuids, startSeqnos)
+	// Initialize the feed based on the backfill type
+	feedInitErr := dcpReceiver.initFeed(args.Backfill)
```

Comment:
The doc comments for `args.Backfill` say:

```
	Backfill   uint64         // Timestamp of oldest item to send. Use TapNoBackfill to suppress all past items.

```

which are a bit confusing.  What updating to comments to say something like:


```
Timestamp of oldest item to send, or: sgbucket.FeedNoBackfill to suppress all past items, sgbucket.FeedResume to resume backfill where left off
```

Reply:
I don't want to pull an sgbucket change into this PR right now, but it's a reasonable enhancement in a separate PR.

Reply:
I just meant the comments, not any actual code change.  Actually it could be shortened to:

> Timestamp of oldest item to send, or: FeedNoBackfill to suppress all past items, FeedResume to resume backfill where left off

Reply:
N/m, I see that `args.Backfill` lives in the sgbucket repo